### PR TITLE
Optimization Pass: Merge Adjacent MaskStmts in OIR

### PR DESCRIPTION
**Pull request by fthaler**
## Description

A new optimization pass, merging adjacent MaskStmts if the masks are identical (and not re-written in the first).

## Requirements

Before submitting this PR, please make sure:

- [x] The code builds cleanly without new errors or warnings
- [x] The code passes all the existing tests
- [x] If this PR adds a new feature, new tests have been added to test these
new features
- [x] All relevant documentation has been updated or added

Additionally, if this PR contains code authored by new contributors:

- [x] All the authors are covered by a valid contributor assignment agreement,
signed by the employer if needed, provided to ETH Zurich
- [x] The names of all the new contributors have been added to an updated
version of the AUTHORS.rst file included in the PR

### CODE REVIEW

**Comment by fthaler** [comment]
bors try

**Comment by fthaler** [comment]
bors try

File: src/gtc/passes/oir_optimizations/mask_stmt_merging.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    def _merge(self, stmts: List[oir.Stmt]) -> List[oir.Stmt]:
+        merged = [self.visit(stmts[0])]
+        for stmt in stmts[1:]:
+            stmt = self.visit(stmt)
+            if (
+                isinstance(stmt, oir.MaskStmt)
+                and isinstance(merged[-1], oir.MaskStmt)
+                and stmt.mask == merged[-1].mask
+                and not (
+                    AccessCollector.apply(merged[-1].body).write_fields()
```
Comment (havogt) [inline src/gtc/passes/oir_optimizations/mask_stmt_merging.py:35]: Please add a test for this branch.

**Review by havogt — CHANGES_REQUESTED** [review]

File: src/gtc/passes/oir_optimizations/mask_stmt_merging.py (hunk 1 above)
Comment (fthaler) [reply src/gtc/passes/oir_optimizations/mask_stmt_merging.py:35]: Done.

**Comment by fthaler** [comment]
bors try

**Comment by fthaler** [comment]
bors try

**Comment by egparedes** [comment]
@fthaler @havogt What is still missing to merge this PR?

**Comment by fthaler** [comment]
> @fthaler @havogt What is still missing to merge this PR?

Just @havogt’s second review round and approval is missing as far as I can tell.

**Review by havogt — APPROVED** [review]

**Comment by fthaler** [comment]
bors try

**Pull request merged**